### PR TITLE
Allow user to edit or delete their proposal

### PR DESCRIPTION
**Pull request by cdt-eth**
## When can you edit/delete?
A user can edit or delete a proposal they've submitted only when the round is in the accepting proposal stage. Once voting in a round begins the proposals are locked. Both updating & deleting a proposal will require a wallet signature for confirmation.

## ***_A note on ties_**
Before, Prop House handled ties by awarding the _earliest submitted_ proposal. However, to prevent gamification, ties will check for editing timestamps versus the prop's creation date. Meaning, if you're in the running for a tie, if you _submitted_ your proposal first but at a later date _edited_ your proposal, that is now the timestamp that will be assessed in determing "who was first."

## Editing a Proposal
When you open your proposal and your wallet is connected you'll see an Edit button. From there and editor will be pre-populated with your current proposal's content. Once you've finished making changes, click "Save," and you'll be prompted to confirm. After signing you'll receive confirmation that it was saved successfully or there was an error and you need to try again. Then you'll be redirected back to the round page and you'll see your updated proposal.

## Deleting a Proposal
It's a similar flow for proposal deletion. Connect your wallet and open a proposal you submitted. Click "Delete" and you'll be prompted by your wallet to sign a message. Once you do you'll see either a success or error modal.

## Modals
modals

### CODE REVIEW

File: packages/prop-house-webapp/src/utils/sortByVotesAndHandleTies.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { StoredProposalWithVotes } from '@nouns/prop-house-wrapper/dist/builders';
+import { getLastUpdatedDate } from './getLastUpdatedDate';
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/utils/sortByVotesAndHandleTies.ts:2]: Nit (please review logic before implementing):

```
const gt = Number(a.voteCount) > Number(b.voteCount);
const eq = Number(a.voteCount) === Number(b.voteCount);
const sortedByUpdated = (dayjs(getLastUpdatedDate(b)) as any) - (dayjs(getLastUpdatedDate(a)) as any);

return eq ? sortedByUpdated : gt ? (ascending ? 1 : -1) : ascending ? -1 : 1;
```

File: packages/prop-house-webapp/src/utils/getWinningIds.ts [hunk 2]
```diff
@@ -12,21 +12,13 @@ const getWinningIds = (
   )
     return;
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/utils/getWinningIds.ts:13]: Should return the same type on both cases. So here we could return an empty array?

File: packages/prop-house-webapp/src/components/DeleteProposalModal/DeleteProposalModal.module.css [hunk 3]
```diff
@@ -0,0 +1,89 @@
+.modal {
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/components/DeleteProposalModal/DeleteProposalModal.module.css:1]: We REALLY REALLY need to refactor all these modals into one. We have this css file repeated prob 5+ times now. 

Let's prioritize after PR merged 🙏

File: packages/prop-house-webapp/src/components/ProposalModal/index.tsx [hunk 4]
```diff
@@ -117,7 +125,7 @@ const ProposalModal = () => {
   }, [handleKeyPress]);
 
   const handleDirectionalArrowClick = (direction: Direction) => {
-    if (!activeProposal || !proposals || proposals.length === 0) return;
+    if (!activeProposal || !proposals || proposals.length === 0 || editProposalMode || showDeletePropModal) return;
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/components/ProposalModal/index.tsx:128]: Very clean!

File: packages/prop-house-webapp/src/components/ProposalModalNavButtons/ProposalModalNavButtons.module.css [hunk 5]
```diff
@@ -47,5 +47,6 @@
 @media (max-width: 575px) {
   .btnContainer {
     padding: 0px 20px 20px 20px;
+    /* padding: 0px 12px; */
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/components/ProposalModalNavButtons/ProposalModalNavButtons.module.css:50]: Remove

File: packages/prop-house-webapp/src/components/SaveProposalModal/SaveProposalModal.module.css [hunk 6]
```diff
@@ -0,0 +1,89 @@
+.modal {
```
Comment (cryptoseneca) [inline packages/prop-house-webapp/src/components/SaveProposalModal/SaveProposalModal.module.css:1]: ☠️☠️☠️☠️☠️☠️☠️☠️☠️

**Review by cryptoseneca — COMMENTED** [review]
*I f'ed up this review w multiple comments/reviews by doing it straight from vs code*

UX Feedback:
- On delete/edit, if we ask for signature but the signature fails for w/e reason (eg decline to sign), we should reflect that action in the UI as we do with voting where we show “failed to submit votes [banana]” modal.
- If data fields on editor modal don’t match - let’s add signifiers as we do in the main create prop page
- Deleting/editing a prop and clicking outside the “successful [action]” modal does not dismiss the prop modal and allows for another attempt to delete/edit the same prop. To reproduce:
1. Open prop modal and navigate to prop you submitted
2. Click “delete” (or edit) and sign message
3. You get the “Successfully Deleted! (or edited)” modal. 
4. **If you click outside of the modal to dismiss the modal**, you will still see the modal and be able to retry the deletion of the prop

File: packages/prop-house-webapp/src/utils/sortByVotesAndHandleTies.ts (hunk 1 above)
Comment (cdt-eth) [reply packages/prop-house-webapp/src/utils/sortByVotesAndHandleTies.ts:2]: Updated with a missing case: https://github.com/Prop-House/prop-house-monorepo/pull/344/commits/7de4faaf2528bcf3bce041a63ffe936de45ed53f

File: packages/prop-house-webapp/src/utils/getWinningIds.ts (hunk 2 above)
Comment (cdt-eth) [reply packages/prop-house-webapp/src/utils/getWinningIds.ts]: returning empty array vs returning nothing & therefore can remove some conditionals in the UI

https://github.com/Prop-House/prop-house-monorepo/pull/344/commits/893a9643011aad2ead350701f4f04b0d7728dd67

File: packages/prop-house-webapp/src/components/DeleteProposalModal/DeleteProposalModal.module.css (hunk 3 above)
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/DeleteProposalModal/DeleteProposalModal.module.css:1]: will do

File: packages/prop-house-webapp/src/components/SaveProposalModal/SaveProposalModal.module.css (hunk 6 above)
Comment (cdt-eth) [reply packages/prop-house-webapp/src/components/SaveProposalModal/SaveProposalModal.module.css:1]: i knoowwwwwwwwww

**Comment by cdt-eth** [comment]
### 1) "On delete/edit, ...signature fails for w/e reason ... we should reflect that action in the UI"
Updated logic for this: https://github.com/Prop-House/prop-house-monorepo/pull/344/commits/b9ae07f5bdcea2b234c147fb441d7df4946444bb

### 2) "If data fields on editor modal don’t match - let’s add signifiers as we do in the main create prop page"
This already exists

### 3) "Deleting/editing a prop and clicking outside the “successful [action]” modal does not dismiss the prop modal and allows for another attempt to delete/edit the same prop."
Done: https://github.com/Prop-House/prop-house-monorepo/pull/344/commits/2ee260023864e173e2c070f44e9589b01a9f628c & https://github.com/Prop-House/prop-house-monorepo/pull/344/commits/5a1dcea940bce1e7ae911803a6f6c006510f6b63

**Review by cryptoseneca — APPROVED** [review]
lgtm! one last thing: deleting the prop and dismissing the "success" modal by clicking outside of the modal still shows deleted prop in round.

**Pull request merged**